Code/Tools/ProjectManager/Source/ProjectExportController.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <ProjectExportController.h>
+#include <ProjectExportWorker.h>
+#include <ProjectButtonWidget.h>
+#include <SettingsInterface.h>
+
+#include <QMessageBox>
+#include <QDebug>
+#include <QDesktopServices>
+#include <QUrl>
+
+namespace O3DE::ProjectManager
+{
+    ProjectExportController::ProjectExportController(const ProjectInfo& projectInfo, ProjectButton* projectButton, QWidget* parent)
+        : QObject()
+        , m_projectInfo(projectInfo)
+        , m_projectButton(projectButton)
+        , m_parent(parent)
+    {
+        m_worker = new ProjectExportWorker(m_projectInfo);
+        m_worker->moveToThread(&m_workerThread);
+
+        // Remove key here in case Project Manager crashed while building because that causes HandleResults to not be called
+        SettingsInterface::Get()->SetProjectBuiltSuccessfully(m_projectInfo, false);
+
+        connect(&m_workerThread, &QThread::finished, m_worker, &ProjectExportWorker::deleteLater);
+        connect(&m_workerThread, &QThread::started, m_worker, &ProjectExportWorker::ExportProject);
+        connect(m_worker, &ProjectExportWorker::Done, this, &ProjectExportController::HandleResults);
+        connect(m_worker, &ProjectExportWorker::UpdateProgress, this, &ProjectExportController::UpdateUIProgress);
+    }
+
+    ProjectExportController::~ProjectExportController()
+    {
+        m_workerThread.requestInterruption();
+        m_workerThread.quit();
+        m_workerThread.wait();
+    }
+
+    void ProjectExportController::Start()
+    {
+        m_workerThread.start();
+        UpdateUIProgress("");
+    }
+
+    void ProjectExportController::SetProjectButton(ProjectButton* projectButton)
+    {
+        m_projectButton = projectButton;
+
+        if (projectButton)
+        {
+            projectButton->SetProjectButtonAction(tr("Cancel"), [this] { HandleCancel(); });
+            projectButton->SetBuildLogsLink(QUrl::fromLocalFile(m_worker->GetLogFilePath()));
+            projectButton->SetState(ProjectButtonState::Exporting);
+
+            if (!m_lastLine.isEmpty())
+            {
+                UpdateUIProgress(m_lastLine);
+            }
+        }
+    }
+
+    const ProjectInfo& ProjectExportController::GetProjectInfo() const
+    {
+        return m_projectInfo;
+    }
+
+    void ProjectExportController::UpdateUIProgress(const QString& lastLine)
+    {
+        m_lastLine = lastLine.left(s_maxDisplayedBuiltOutputChars);
+
+        if (m_projectButton)
+        {
+            m_projectButton->SetContextualText(m_lastLine);
+        }
+    }
+
+    void ProjectExportController::HandleResults(const QString& result)
+    {
+        if (!result.isEmpty())
+        {
+            if (result.contains(tr("log")))
```

Comment:
I can see potential errors here. If the result is actually not a log file, but the error says something like 'Error creating log file'. then you don't actually have a log file. What are the possible values of `result` ?

Reply:
This function is only called from `ProjectExportWorker::ExportProject`, via the emitting of the `Done` signal. In the case of failure the result error message is always fetched, which is a Qstring message.

This code was also part of the builder code, and I think at the time it was based on the qstring error "Building project failed. See log for details.". Chances are that at its inception, this was the only message at the time, which made it easy to select, but as the code grew more complex this got overlooked.

Reply:
In this case though I dont think the error is significant, since in either branch the message box should either show the log file path, or have a dialog box that directly opens it. In either case the intended log is accessible in some form.

Reply:
I think rewriting this code snippet should be done in conjunction with the builder code, which is similarly out of date, but that kind of rewrite is out of scope for the current PR

Reply:
Is there a chance the log file will not be created (out of room or anti-virus blocking it) and therefore the dialog box would be empty instead of showing a path?

Reply:
The log might be empty, but the path for where the log would be written is hardcoded in the project manager codebase.

In my testing I have not run into the case where anti-virus interfered in such a manner, but I suspect it's because the log is written via a subordinate process limited to only o3de related folders, so it's possible it's not being considered strange behavior.

Reply:
So does the `log` check meant to determine if the string is the full path of the log? If so, then a check if the result ends with `.log` should be safer